src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -347,6 +352,11 @@ public void MoveTo(String destinationEntryName)
 
         #region Privates
 
+        // Only allow opening ZipArchives with large ZipArchiveEntries in update mode when running in a 64-bit process.
+        // This is for compatibility with old behavior that threw an exception for all process bitnesses, because this
+        // will not work in a 32-bit process.
+        private bool AllowLargeZipArchiveEntriesInUpdateMode { get { return Marshal.SizeOf(new IntPtr()) >= 4; } }
```

Comment:
It'd be nice to put this into a readonly static field.  The JIT will then be able to optimize away branches based on the result.